executor/index_lookup_join.go

```diff
@@ -473,6 +475,10 @@ func (iw *innerWorker) handleTask(ctx context.Context, task *lookUpJoinTask) err
 func (iw *innerWorker) constructLookupContent(task *lookUpJoinTask) ([]*indexJoinLookUpContent, error) {
 	lookUpContents := make([]*indexJoinLookUpContent, 0, task.outerResult.NumRows())
 	keyBuf := make([]byte, 0, 64)
+	keyTypes := make([]*types.FieldType, 0, len(iw.keyCols))
+	for _, pos := range iw.keyCols {
+		keyTypes = append(keyTypes, iw.rowTypes[pos])
```

Comment:
What about not saving `keyType`, but use `iw.rowTypes[iw.keyCols[i]]` instead in line 503.

Reply:
ok